api/src/pcapi/core/offerers/api.py

```diff
@@ -2321,5 +2323,54 @@ def match_venue_with_new_entries(
 
 
 def update_offerer_address_label(offerer_address_id: int, new_label: str) -> None:
-    models.OffererAddress.query.filter_by(id=offerer_address_id).update({"label": new_label})
-    db.session.flush()
+    try:
+        models.OffererAddress.query.filter_by(id=offerer_address_id).update({"label": new_label})
+        db.session.flush()
+    except sa.exc.IntegrityError:
+        raise exceptions.OffererAddressLabelAlreadyUsed()
+
+
+def get_or_create_address(address_info: AddressInfo) -> geography_models.Address:
+    try:
+        address = geography_models.Address(
+            banId=address_info.id,
+            inseeCode=address_info.citycode,
+            street=address_info.street,
+            postalCode=address_info.postcode,
+            city=address_info.city,
+            latitude=address_info.latitude,
+            longitude=address_info.longitude,
+        )
+        db.session.add(address)
+        db.session.flush()
+    except sa.exc.IntegrityError:
+        db.session.rollback()
+        address = geography_models.Address.query.filter(
+            geography_models.Address.street == address_info.street,
+            geography_models.Address.inseeCode == address_info.citycode,
+        ).one()
+        if (address.latitude, address.longitude) != (address_info.latitude, address.longitude):
+            logger.error("Unique contraint over street and inseeCode matched different coordinates")
```

Comment:
```suggestion
            logger.error("Unique constraint over street and inseeCode matched different coordinates", extra={...})
```
Je pense qu'il faut aider au debug, avec ce qui est demandé et l'id en base de données.

Reply:
Dans tous les cas ça popera dans Sentry et on aura accès au contexte, mais j'ai rajouté l'id de l'adresse et le banId de l'API